File: c++/test1.cpp
#include <iostream>
#include <vector>
using namespace std;
int main()
{
  vector<int> nums;
  int num;
  while (getchar() != ']')
  {
    cin >> num;
    nums.push_back(num);
  }
  int size = nums.size();
  int i = 0, a = nums[0];
  while (1)
  {
    i += a;
    cout << "*****: " << i << " " << a << endl;
    if (i < 0 || i >= size)
    {
      cout << "true";
      return 0;
    }
    if (nums[i] == 0)
    {
      cout << "false";
      return 0;
    }
    a = nums[i];
    nums[i] = 0;
  }
  return 0;
}

File: leetcode+牛客/test1.py
def quickSort(nums): return nums if len(nums) < 2 else quickSort([i for i in nums[1:] if i < nums[0]]) +\
    [nums[0]]+quickSort([c for c in nums[1:] if c >= nums[0]])


nums = [1, 4, 5, 2, 1, 1, 2, 1, 3]
print(quickSort(nums))
